packages/pigeon/platform_tests/test_plugin/ios/Classes/CoreTests.gen.swift

```diff
@@ -20,11 +20,11 @@ private func wrapResult(_ result: Any?) -> [Any?] {
   return [result]
 }
 
-private func wrapError(_ error: FlutterError) -> [Any?] {
+private func wrapError(_ error: Error) -> [Any?] {
   return [
-    error.code,
-    error.message,
-    error.details
+    "\(type(of: error))",
+    "\(error)",
+    "\(Thread.callStackSymbols)"
```

Comment:
That's correct. I think there are probably better options I could be passing in here, especially for the 'code' section.

Reply:
Is `PlatformException.code` the first or second item in the list?

Reply:
you can also cast it to NSError and get the code (which will be an integer starting from 0). Does the dart side automatically generate the error type? or do developers have to manually translate the code? Can you show me how this is used in Dart?

Reply:
> Is `PlatformException.code` the first or second item in the list?

1. code
2. message
3. details

Reply:
From the doc, the platform exception has separate details and stacktrace: 

[PlatformException](https://api.flutter.dev/flutter/services/PlatformException/PlatformException.html)({required [String](https://api.flutter.dev/flutter/dart-core/String-class.html) code, [String](https://api.flutter.dev/flutter/dart-core/String-class.html)? message, dynamic details, [String](https://api.flutter.dev/flutter/dart-core/String-class.html)? stacktrace})

Is it intended to make the `callstack` the `details`?

Reply:
I took another look of this PR. If I understand it correctly, you changed `FlutterError` to `Error` because `do/try/catch` only gives you `Error` type? 

I think it's still better to throw `FlutterError` since it contains `code` and `message` which is far more helpful. A workaround can be just casting it, with a fallback to general errors. Something like: 

```
  if let flutterError = error as? FlutterError {
    return [
      flutterError.code,
      flutterError.message,
      flutterError.details,
    ]
  } else {
    // fallback to general errors like you did here
  }
```

Reply:
> I think it's still better to throw `FlutterError` since it contains `code` and `message` which is far more helpful. 

Can swift `throw` a FlutterError? Is that something that's possible?

Reply:
you can do something like

```
extension FlutterError: Error {}
```